web/impact/impact/v1/views/user_program_participation_view.py

```diff
@@ -0,0 +1,106 @@
+from django.contrib.auth import get_user_model
+from django.db.models import F
+from django.template import loader
+from rest_framework.exceptions import ParseError
+from rest_framework.response import Response
+from accelerator.models import (
+    UserRole,
+    Program,
+    ProgramRoleGrant,
+    ProgramRole
+)
+from impact.minimal_email_handler import MinimalEmailHandler
+from impact.permissions.v1_api_permissions import IsExpertUser
+from impact.v1.views import ImpactView
+from impact.v1.views.user_detail_view import NO_USER_ERROR
+
+User = get_user_model()
+
+SUBJECT = '[MassChallenge] Confirmed: Your Mentorship Commitment'
+INVALID_INPUT_ERROR = "'{}' must be a list of numbers."
+
+
+def extract_values(request, key):
+    try:
+        return [int(i) for i in request.data.get(key, [])]
+    except ValueError:
+        raise ParseError(INVALID_INPUT_ERROR.format(key))
+
+
+class ExpertParticipationView(ImpactView):
```

Comment:
I notice that the view name and the class name are out of sync. I personally think `MentorParticipationView` is more apt than `ExpertParticipationView`, but I would be willing to accept the latter - in any case, the view_name should conform to the class name, and the file name should also agree with both.

Reply:
FWIW this view will be used for the foreseeable future for mentor confirmation only. There is a desire, but no current plan, to have a more similar (possibly the same) workflow for all types of expert confirmation (judge and mentor), so you may consider that when deciding on a naming convention.

Reply:
switched to `MentorParticipationView`